src/sagemaker/fw_utils.py

```diff
@@ -1066,31 +1074,43 @@ def validate_torch_distributed_distribution(
         return
 
     err_msg = ""
+    # Check instance compatibility
+    if not _is_gpu_instance(instance_type):
+        err_msg += "torch_distributed is supported only for GPU instances.\n"
+
     if not image_uri:
         # ignore framework_version and py_version if image_uri is set
         # in case image_uri is not set, then both are mandatory
-        if framework_version not in TORCH_DISTRIBUTED_SUPPORTED_FRAMEWORK_VERSIONS:
+        if (
+            framework_version not in TORCH_DISTRIBUTED_SUPPORTED_FRAMEWORK_VERSIONS
+            and framework_version not in TORCH_DISTRIBUTED_TRAINIUM_SUPPORTED_FRAMEWORK_VERSIONS
+        ):
             err_msg += (
                 f"Provided framework_version {framework_version} is not supported by"
                 " torch_distributed.\n"
                 "Please specify one of the supported framework versions:"
-                f" {TORCH_DISTRIBUTED_SUPPORTED_FRAMEWORK_VERSIONS} \n"
+                f"{TORCH_DISTRIBUTED_TRAINIUM_SUPPORTED_FRAMEWORK_VERSIONS} \n"
```

Comment:
This should be `TORCH_DISTRIBUTED_SUPPORTED_FRAMEWORK_VERSIONS` for GPU instances. Let's construct this message based on the instance type?

Reply:
updated